sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -318,24 +317,13 @@ def delete_entity(self, *args: Union[TableEntity, str], **kwargs) -> None:
             if row_key is None:
                 row_key = args[1]
 
-        match_condition = kwargs.pop("match_condition", None)
-        etag = kwargs.pop("etag", None)
-        if match_condition and entity and not etag:
-            try:
-                etag = entity.metadata.get("etag", None)
-            except (AttributeError, TypeError):
-                pass
-        if_match = _get_match_headers(
-            etag=etag,
-            match_condition=match_condition or MatchConditions.Unconditionally,
-        )
-
         try:
             self._client.table.delete_entity(
                 table=self.table_name,
                 partition_key=_prepare_key(partition_key),
                 row_key=_prepare_key(row_key),
-                if_match=if_match,
+                etag=kwargs.pop("etag", None),
+                match_condition=kwargs.pop("match_condition", None) or MatchConditions.IfPresent,
```

Comment:
This current code will be breaking - so we will either need to discuss the addition of `Unconditionally` to the generated handlers, or continue to parse these values ourselves.

Reply:
Changed to pass `IfPresent` when "match_condition" is in value `Unconditionally`.